wordpress-mobile/WordPress-iOS
# Lock Screen Widgets: Add Tracks

Fixes #21531 

Adding tracking for all lock screen widgets. Expanding current tracking implementation which tracks when widget is newly added. 

As part of this PR, refactoring configuration so all the different widget configurations would be defined in `enum`. This gives the benefit of the compiler notifying when places relying on this `enum` need to be implemented. Now, when a new widget `Kind` will be added, `ExtensionEvents` will ask to provide additional track events.

### To test:

I used these steps for testing:

1. Launch Console app
2. Action -> Enable Info messages
3. Edit `FeatureFlag.swift` to return `true`  for `.lockScreenWidget`
4. Launch Jetpack
5. Login
6. Open Lock Screen
7. Add Jetpack widget
8. Confirm the event is logged in Console app

## Regression Notes
1. Potential unintended areas of impact

Breaking current widget implementation. However, I didn't change any existing keys, only where they are stored.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
